sage.py

```diff
@@ -481,483 +449,311 @@ def getepisodes(action_seq, mcat, plot, debug=False):
-            plt.plot(xax_end, yax, 'r', linestyle=(0, (5, 10)))
-
-        plt.show()
-    #print('number episodes ', len(episodes_))
-    return episodes_
-
-def aggregate_into_episodes(unparse, team_labels, step=150):
-    cols = ['b', 'r', 'g', 'c', 'm', 'y', 'k', 'olive', 'lightcoral', 'skyblue', 'mediumpurple', 'springgreen', 'chocolate', 'cadetblue', 'lavender']
+        _plot_episodes(frequencies, episodes, mcat)
 
-    PRINT = False
-    interesting = []
-    # Reorganize data for each attacker per team 
-    team_data = dict()
-    s_t = dict()
-    for tid,team in enumerate(unparse):
-        #attackers = list(set([x[1] for x in team])) # collecting all src ip
-        #attackers.extend(list(set([x[3] for x in team]))) # collection all dst ip
-        #attackers = [x for x in attackers if x not in hostip.keys()] # filtering only attackers
-        
-        
-        host_alerts = dict()
-
-        for ev in team:
-            #print(ev[0])
-            h = ev[7]
-            s = ev[1]
-            d = ev[3]
-            c = ev[9]
-            ts = ev[8]
-            sp = ev[2] if ev[2] != None else 65000 
-            dp = ev[4] if ev[4] != None else 65000
-            signature = ev[5]
-            # Simply respect the source,dst format! (Correction: source is always source and dest alwyas dest!)
-            
-            source, dest, port = -1, -1, -1
-            #print(s, d, sp, dp)
-            #assert sp >= dp
-            
-            source = s #if s not in inv_hostip.keys() else inv_hostip[s]
-            dest = d #if d not in inv_hostip.keys() else inv_hostip[d]
-            # explicit name if cant resolve
-            #port = str(dp) if (dp not in port_services.keys() or port_services[dp] == 'unknown') else port_services[dp]['name'] 
-            
-            # say unknown if cant resolve it
-            port = 'unknown' if (dp not in port_services.keys() or port_services[dp] == 'unknown') else port_services[dp]['name']
-            
-            if (source,dest) not in host_alerts.keys() and (dest,source) not in host_alerts.keys():
-                host_alerts[(source,dest)] = []
-                #print(tid, (source,dest), 'first', ev[8])
-                s_t[str(tid)+"|"+str(source)+"->"+str(dest)] = ev[8]
-                
-            if((source,dest) in host_alerts.keys()):
-                host_alerts[(source,dest)].append((dest, c, ts, port, signature)) # TODO: remove the redundant host names
-                #print(source, dest, (micro[c].split('.'))[-1], port)
-            else:
-                host_alerts[(dest,source)].append((source, c, ts, port, signature))
-                #print(dest,source, (micro[c].split('.'))[-1], port)
+    return episodes
 
-        team_data[tid] = host_alerts.items()
-        
-    # Calculate number of alerts over time for each attacker 
-    #print(len(s_t))
-    team_episodes = []
 
+def _create_episode(alert_seq_epi, mcat, tid):
```

Comment:
This function is essentially a compact version of all the long list comprehensions from the `aggregate_into_episodes` function

Reply:
I hope you kept the commented out test cases still in the code.

Reply:
Yes, but they are in `_get_episodes` method. I have added a TODO that they could be moved to a separate test file.